onnx/helper.py

```diff
@@ -482,6 +486,59 @@ def make_tensor_value_info(
     return value_info_proto
 
 
+def make_sparse_tensor_value_info(
+        name,  # type: Text
+        elem_type,  # type: int
+        shape,  # type: Optional[Sequence[Union[Text, int]]]
+        doc_string="",  # type: Text
+        shape_denotation=None,  # type: Optional[List[Text]]
+):  # type: (...) -> ValueInfoProto
+    """Makes a ValueInfoProto based on the data type and shape."""
+    value_info_proto = ValueInfoProto()
+    value_info_proto.name = name
+    if doc_string:
+        value_info_proto.doc_string = doc_string
+
+    sparse_tensor_type_proto = value_info_proto.type.sparse_tensor_type
+    sparse_tensor_type_proto.elem_type = elem_type
+
+    sparse_tensor_shape_proto = sparse_tensor_type_proto.shape
```

Comment:
I think we should move this into the if-then branch? Otherwise, we could create a rank-zero (scalar) instead of an unknown-rank tensor for the shape=None case?

Reply:
i dont think so... if shape is None as explained in the comment below this field will be omitted from the resulting protobuf... 

we take a similar approach for tensor proto as well

Reply:
yes, the fact that is is used for tensor suggests it works. I am still unclear what exactly happens. It is possible that a "shape" object is created (when the above statement is executed), and that when it is saved to protobuf it is omitted because it has no fields? There is a difference between the "shape" object and its field "dims" ... the comment below describes whether a dims field is created or not. But my question is about whether the "shape" field is created or not, since in C++ we do check for the existence of the "shape" field to determine if it has a rank or not.

Reply:
Nevertheless, wouldn't creating the shape field inside the if-then more accurately reflect what we want?

Reply:
Ok, I tried it out. It seems to work. It is a bit mysterious how exactly the protobuf implementation works.

Reply:
Yes, it creates a shape object, that is empty. It is the same as calling mutable_shape() in C++

Reply:
Ok, this seems to work. This line does not actually create a shape field. A bit mysterious, but seems ok (just python-approach to protobuf).